src/Keystore/StoredKey.cpp

```diff
@@ -114,14 +114,31 @@ const PrivateKey StoredKey::privateKey(TWCoinType coin, const std::string& passw
 }
 
 void StoredKey::fixAddresses(const std::string& password) {
-    const auto wallet = this->wallet(password);
-    for (auto& account : accounts) {
-        if (!account.address.empty() && TW::validateAddress(account.coin(), account.address)) {
-            continue;
+    switch (type) {
+    case StoredKeyType::mnemonicPhrase: {
+        const auto wallet = this->wallet(password);
+        for (auto& account : accounts) {
+            if (!account.address.empty() && TW::validateAddress(account.coin(), account.address)) {
+                continue;
+            }
+            const auto& derivationPath = account.derivationPath;
+            const auto key = wallet.getKey(derivationPath);
+            account.address = TW::deriveAddress(derivationPath.coin(), key);
         }
-        const auto& derivationPath = account.derivationPath;
-        const auto key = wallet.getKey(derivationPath);
-        account.address = TW::deriveAddress(derivationPath.coin(), key);
+    }
+        break;
+    case StoredKeyType::privateKey: {
+        auto key = PrivateKey(payload.decrypt(password));
+        for (auto& account : accounts) {
+            if (!account.address.empty() && TW::validateAddress(account.coin(), account.address)) {
+                continue;
+            }
+            account.address = TW::deriveAddress(account.coin(), key);
+        }
+    }
+        break;
+    case StoredKeyType::watchOnly:
+        break;
```

Comment:
should we do validation here? Keystore.watchOnly doesn't use by ios app yet.

Reply:
just list all cases here, without private key, we can't "fix" an address